### DISCUSSION TITLE
accessToken을 어떻게 관리하는것이 안전할까요?

### DISCUSSION THREAD

**Discussion by airman5573** [woowacourse-study/2022-woowahan-bansanghwe#97] [Network]
accessToken은 쿠키에 저장하는것이 적절합니다.  

## 왜죠? 
- 프론트단에서 따로 컨트롤 하지 않아도 매 요청마다 accessToken이 전송되기 때문에 편합니다.  
- httpOnly옵션을 적용할시 javascript단에서 쿠키에 접근할 수 없기 때문에 XSS공격을 막을 수 있습니다.  

## localStorage에 저장하면 안되는 이유가 있을까요?  
localStorage는 XSS공격에 취약하기 때문에 안전하지 않습니다.

## session은요?  
브라우저를 껐을때 세션정보가 날라가기 때문에, 사용자는 다시 로그인을 해야합니다.  
그리고 localStorage와 마찬가지로 XSS공격에 취약합니다.

## javascript메모리에 저장하는 방법은 어떤가요?  
cookie와 마찬가지로 XSS공격을 막을 수는 있지만, 새로고침을 할때마다 다시 accessToken을 서버로부터 받아와야 하는 단점이 있습니다.

**Comment by al-bur** [comment]
왜 이 주제를 선정하고 계셨죠?

**Comment by al-bur** [comment]
accessToken이 뭔가요?

**Comment by al-bur** [comment]
accessToken과 JWT는 다른 건가요?

**Comment by al-bur** [comment]
JWT는 뭐죠?

**Reply by al-bur** [reply]
[꼬리질문] JWT는 인증 인가에 국한된 건가요?

**Comment by al-bur** [comment]
해커에게 탈취당했을 때 어떻게 하면 조금이라도 도움을 줄 수 있을지?

**Reply by al-bur** [reply]
[꼬리질문] 어떻게 네이버는 일회용 로그인을 구현을 했을까요?

**Comment by byhhh2** [comment]
- httpOnly 옵션을 적용했다고 해도 네트워크 탭에서 획득하면 되지 않나?
    - 이런 경우 최선의 해결은?

**Comment by byhhh2** [comment]
refreshToken의 사용 방법?

**Comment by al-bur** [comment]
cookie의 헤더에 매번 들어가는 것에 대한 트레이드오프?

**Comment by al-bur** [comment]
session 방식을 사용하지 않고 토큰을 사용한 이유는?

**Comment by byhhh2** [comment]
크롬에서 same-site default값이 바뀌면서 cross-site간 cookie를 주고 받기 어려워졌는데 이를 해결한 경험이 있는가?

**Comment by byhhh2** [comment]
- XSS 공격이란?
    - 쿠키와의 관계는

**Comment by al-bur** [comment]
react에서 xss 공격을 막아준다고 하셨는데, xss 공격을 막아주기 위해(이유들 중 하나) cookie를 사용하셨다고 하신 것에 대해서는 어떻게 생각하시나요?

**Comment by al-bur** [comment]
재조정이 무엇인가요?

**Reply by byhhh2** [reply]
재조정의 알고리즘은?

**Comment by byhhh2** [comment]
- HTTP 단에서의 보안적인 해결은?

**Reply by byhhh2** [reply]
HTTPS란?
